tests/system/test_crud.py

```diff
@@ -1429,3 +1429,32 @@ class SomeKind(ndb.Model):
 
     retrieved = key.get()
     assert retrieved.foo == datetime.datetime(2020, 8, 8, 1, 2, 3)
+
+
+@pytest.mark.usefixtures("client_context")
+def test_cache_returns_entity_if_available(dispose_of, client_context):
+    """Regression test for #441
+
+    https://github.com/googleapis/python-ndb/issues/441
+    """
+
+    class SomeKind(ndb.Model):
+        foo = ndb.IntegerProperty()
+        bar = ndb.StringProperty()
+
+    class OtherKind:
+        def __init__(self, somekind):
+            self.somekind = somekind
+
+    client_context.set_cache_policy(None)  # Use default
+
+    somekind = SomeKind(foo=1)
+    key = somekind.put()
+    dispose_of(key._key)
+    otherkind = OtherKind(somekind)
```

Comment:
What does this bit of indirection get us?

Reply:
I was trying to follow the user's sample code. I'll see if I can simplify that.